basic_flutter/lib/data/network/http_client_provider.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:dio/dio.dart';
+import 'package:pretty_dio_logger/pretty_dio_logger.dart';
+
+class HttpClientProvider {
```

Comment:
Why does this class exists?

Reply:
I read the class and it's there for the possibility of adding and authentication system to the whole application. It handles the authorization of the incoming token and how it responds to the jwt token

Reply:
Did you write the class? For me its seems like a class thats just not used. If you think you will need it later describe what it does and for what it can be used.

Reply:
No, like what was mentioned in the comment from Alessandro, we got these data models from the previous flutter project. It's not used because neither in the previous project nor this project have we developed an authentication system. I'll add the description to the top of the class and leave it there, we will use it once we make a final decision.